packages/yoga/src/Button/web/Button.jsx

```diff
@@ -13,20 +13,30 @@ const Button = ({
   secondary,
   icon: Icon,
   ...props
-}) => (
-  <StyledButton
-    disabled={disabled}
-    full={full}
-    inverted={inverted}
-    onClick={onClick}
-    small={small}
-    secondary={secondary}
-    {...props}
-  >
-    {Icon && <Icon />}
-    {children}
-  </StyledButton>
-);
+}) => {
+  const finalProps = {
```

Comment:
It's just a suggestion, but if think it's better to create a specific variable to handle the "as" prop instead of creating another object with the props and mutating it.

`
  const ButtonComponent = 'button';
  const LinkComponent = 'a';
 
  let asComponent = ButtonComponent;
  
  if (props.href) asComponent = LinkComponent;
  
  return (
    <StyledButton
      disabled={disabled}
      full={full}
      inverted={inverted}
      onClick={onClick}
      small={small}
      secondary={secondary}
      {...props}
      as="asComponent"
    >
      {Icon && <Icon />}
      {children}
    </StyledButton>
  );
`